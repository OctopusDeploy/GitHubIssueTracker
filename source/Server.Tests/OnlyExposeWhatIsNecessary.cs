using System;
using System.Linq;
using Assent;
using NUnit.Framework;

namespace Octopus.Server.Extensibility.IssueTracker.GitHub.Tests
{
    [TestFixture]
    public class OnlyExposeWhatIsNecessary
    {
        [Test]
        public void ServerExtensionsShouldMinimiseWhatIsExposed()
        {
            var assembly = typeof(GitHubIssueTrackerExtension).Assembly;

            var publicThings = assembly.GetExportedTypes()
                .Select(t => t.FullName);
            
            this.Assent(string.Join(Environment.NewLine, publicThings));
        }
    }
}