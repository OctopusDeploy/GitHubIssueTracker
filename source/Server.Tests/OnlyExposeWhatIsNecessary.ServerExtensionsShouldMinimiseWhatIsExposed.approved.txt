Octopus.Server.Extensibility.IssueTracker.GitHub.GitHubIssueTrackerExtension